### PR TITLE
Sites Dashboard: Second pass at Sites Table design

### PR DESCRIPTION
**Pull request by danielbachhuber**
See https://github.com/Automattic/wp-calypso/issues/65166

## Proposed Changes

A second pass at design for the Sites Table:
* Abstracts the site table row to a `SitesTableRow` component.
* Styling pass on the table row cells.
* Links the site icon and adds `title=""` attributes to links.
* Adds an `EllipsisMenu` with a sample menu item.

image

**This pull request intentionally does not address all items on the linked issue.** To avoid blocking other team members, I'd like to iterate on the issue over 2-4 pull requests. Please limit feedback to the code submitted on the pull request.

## Testing Instructions

1. Check out the branch locally.
2. Navigate to `/sites-dashboard` and view the new Sites Dashboard.

### CODE REVIEW

File: client/sites-dashboard/components/sites-table-row.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		<PopoverMenuItem href={ getDashboardUrl( site.slug ) }>
+			{ __( 'Visit Dashboard' ) }
+		</PopoverMenuItem>
+	);
+};
+
+export default function SitesTableRow( { site }: SiteTableRowProps ) {
+	const { __ } = useI18n();
+	return (
+		<Row key={ site.ID }>
```
Comment (vykes-mac) [inline client/sites-dashboard/components/sites-table-row.tsx]: non-blocking: What's the use of this key? sorting purposes?

**Review by vykes-mac — APPROVED** [review]

File: client/sites-dashboard/components/sites-table-row.tsx (hunk 1 above)
Comment (danielbachhuber) [reply client/sites-dashboard/components/sites-table-row.tsx]: > non-blocking: What's the use of this key? sorting purposes?

@vykes-mac Great catch! The `key=` is actually unnecessary in this context — it was copy-pasta. I've removed it in 6ef1d7ca7e

Regarding the use of `key=` generally, [here's my understanding](https://reactjs.org/docs/lists-and-keys.html#keys):

> Keys help React identify which items have changed, are added, or are removed. Keys should be given to the elements inside the array to give the elements a stable identity

We are using `key=` for our use of the `<SitesTableRow>` component:

https://github.com/Automattic/wp-calypso/blob/6ef1d7ca7e7df3fae7253d443e43ebb46be915ec/client/sites-dashboard/components/sites-table.tsx#L48-L52

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7422105

Thank you @danielbachhuber for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7422105).

**Pull request merged**